UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -234,12 +225,14 @@ void OnDisable()
             m_Brain?.Dispose();
         }
 
-        void NotifyAgentDone()
+        void NotifyAgentDone(bool maxStepReached = false)
         {
             m_Info.done = true;
+            m_Info.maxStepReached = maxStepReached;
             // Request the last decision with no callbacks
             // We request a decision so Python knows the Agent is disabled
             m_Brain?.RequestDecision(m_Info, sensors, (a) => { });
+            m_EpisodeId = EpisodeIdCounter.GetEpisodeId();
```

Comment:
Explain why we need to change the ID here.

Reply:
Added a comment